tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py

```diff
@@ -181,7 +178,20 @@ def create_run_config(
     run_azuremlcompute = RunConfiguration()
     run_azuremlcompute.target = cpu_cluster
     run_azuremlcompute.environment.docker.enabled = True
-    run_azuremlcompute.environment.docker.base_image = docker_proc_type
+    # See https://learn.microsoft.com/en-us/azure/machine-learning/how-to-train-with-custom-image?view=azureml-api-1#use-a-custom-dockerfile-optional
```

Comment:
@simonzhaoms right now in the actions we are installing `run: pip install --quiet "azureml-core>1,<2" "azure-cli>2,<3"`. What is the azurml sdk that we are installing? Here I see the latest one as 1.51 https://pypi.org/project/azureml-sdk/#history.

Of the 3 options, I think one that is interesting to explore would be `Try install everything in the docker file without using Conda.` iif we are reducing dependencies. I think 80% of our problems come from dependencies: https://github.com/microsoft/recommenders/issues/1936 So maybe something to reflect on is how can we reduce dependencies and use more standardize and robust software?

Reply:
@miguelgfierro 

> @simonzhaoms right now in the actions we are installing run: pip install --quiet "azureml-core>1,<2" "azure-cli>2,<3". What is the azurml sdk that we are installing? Here I see the latest one as 1.51 https://pypi.org/project/azureml-sdk/#history.

I think this `azureml-core` is only used for launching the script [`submit_groupwise_azureml_pytest.py`](https://github.com/microsoft/recommenders/blob/simonz-dep-upgrade-20230606/tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py).

https://github.com/microsoft/recommenders/blob/b71c4ed66991f8eddfd80a8afbff05b34591c7ee/.github/actions/azureml-test/action.yml#L75-L77

And the AzureML SDK I mentioned is used inside the docker image launched inside `submit_groupwise_azureml_pytest.py`

https://github.com/microsoft/recommenders/blob/b71c4ed66991f8eddfd80a8afbff05b34591c7ee/.github/actions/azureml-test/action.yml#L85-L94

https://github.com/microsoft/recommenders/blob/b71c4ed66991f8eddfd80a8afbff05b34591c7ee/tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py#L178-L194

I think when we use `CondaDependencies.add_pip_package("xxx")`, AzureML adds the item `xxx` in a Conda env yaml file maintained by itself, and the AzureML SDK is also an item added by AzureML by default implicitly.

https://github.com/microsoft/recommenders/blob/b71c4ed66991f8eddfd80a8afbff05b34591c7ee/tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py#L202-L223

What I don't know is why the AzureML SDK trigger the error now when I try to add support for Python 3.10 and upgrade all dependencies.

Reply:
> Of the 3 options, I think one that is interesting to explore would be Try install everything in the docker file without using Conda. iif we are reducing dependencies. I think 80% of our problems come from dependencies: https://github.com/microsoft/recommenders/issues/1936 So maybe something to reflect on is how can we reduce dependencies and use more standardize and robust software?

I agree.  In addition, if possible, I'd use what GitHub actions and workflows can provide to build the testing pipeline rather than use the AzureML service, because AzureML service is not transparent.